doc/nrf/app_dev/device_guides/nrf54h/ug_nrf54h20_suit_recovery.rst

```diff
@@ -236,3 +236,181 @@ To turn an application into a recovery application, the following steps have to
+This ensures that the SUIT manifest is processed further by the Secure Domain Firmware, allowing the main application to boot.
+
+To enable this feature, the following steps must be performed.
+This assumes that the companion image is orchestrated by the ``APP_LOCAL_3`` SUIT manifest, which is the default setup when using the recovery firmware as the companion.
+
+1. Add the appropriate checking code to the recovery (companion) application in its startup code.
+   Use the following code snippet as a reference:
+
+   .. code-block:: c
+
+      static int should_enter_recovery_check(void)
+      {
+         suit_boot_mode_t mode = SUIT_BOOT_MODE_INVOKE_RECOVERY;
+         suit_ssf_err_t err = SUIT_PLAT_SUCCESS;
+         int ret = 0;
+
+         err = suit_boot_mode_read(&mode);
+
+         if (err != SUIT_PLAT_SUCCESS) {
+            suit_invoke_confirm(-EPIPE);
+            return -EPIPE;
+         }
+
+         if (mode == SUIT_BOOT_MODE_INVOKE) {
+            if (/* add the condition here */) {
+               err = suit_foreground_dfu_required();
+            }
+         }
+
+         if (err != SUIT_PLAT_SUCCESS) {
+            ret = -EPIPE;
+         }
+
+         (void)suit_invoke_confirm(ret);
+
+         return ret;
+      }
+
+      SYS_INIT(should_enter_recovery_check, APPLICATION, CONFIG_APPLICATION_INIT_PRIORITY);
+
+   The code should be placed in the recovery (companion) application's source code.
+   To implement this, you can do one of the following:
+
+   * Create a module with the code snippet and integrate it into the Nordic-provided recovery application.
+   * Create a custom recovery application or companion image and include the module there.
+
+2. If using the default SUIT manifests and running the recovery image as the companion image, no modifications to the manifest templates are needed.
+   Instead, set the :kconfig:option:`CONFIG_SUIT_INVOKE_APP_LOCAL_3_BEFORE_MAIN_APP` option in the recovery image.
+   This can be done by adding ``-Drecovery_CONFIG_SUIT_INVOKE_APP_LOCAL_3_BEFORE_MAIN_APP=y`` to the build command when building the recovery application.
+
+   However, otherwise, the manifest templates have to be modified as follows:
+
+a. In the APP_LOCAL_3 manifest template, add the following code before the ``suit-directive-invoke`` directive on the companion image:
+
+   .. code-block:: yaml
+
+      - suit-directive-override-parameters:
+          suit-parameter-invoke-args:
+            suit-synchronous-invoke: True
+            suit-timeout: 1000
+
+b. In the root manifest template add the ``INSTLD_MFST`` component for the local manifest orchestrating the comapanion image.
+   Make sure it is invoked before the main application images.
+   If the modified template is based on the root manifest template from |NCS|, it will in most cases be enough to add it as the first manifst in the component list.
+   The following code snippet is responsible for this in the default root manifest template:
```

Comment:
```suggestion
     .. code-block:: yaml
  
        - suit-directive-override-parameters:
            suit-parameter-invoke-args:
              suit-synchronous-invoke: True
              suit-timeout: 1000

  b. In the root manifest template, add the ``INSTLD_MFST`` component for the local manifest orchestrating the comapanion image.
     You must ensure it is invoked before the main application images.
     If the modified template is based on the root manifest template from the |NCS|, it will in most cases be enough to add it as the first manifest in the component list.
     The following code snippet is responsible for this in the default root manifest template:
```